Add python3-mlflow to rosdep

n-meunier opened the pull request:
<!-- Thank you for contributing a change to the rosdistro. There are two primary types of submissions.
Please select the appropriate template from below: ROSDEP_RULE_TEMPLATE or DOC_INDEX_TEMPLATE

If you're making a new release with bloom please use bloom to create the pull request automatically.
If you've already run the release bloom has a `--pull-request-only` option you can use.-->


<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

Please add the following dependency to the rosdep database.

## Package name:

`python3-mlflow-pip`
Edit: since it is not only available with `pip`, new name is `python3-mlflow`.

## Package Upstream Source:

https://pypi.org/project/mlflow/
https://build.opensuse.org/package/show/home:jayvdb:py-new/python-mlflow

## Purpose of using this:

https://pypi.org/project/mlflow/#description

A platform to manage Machine Learning models and experiments.

Distro packaging links:

## Links to Distribution Packages

<!-- Replace the REQUIRED areas and state not available for IF AVAILABLE -->

- Debian: https://packages.debian.org/
  - NOT_FOUND
- Ubuntu: https://packages.ubuntu.com/
  - NOT_FOUND
- Fedora: https://src.fedoraproject.org/browse/projects/
  - NOT_FOUND
- Arch: https://www.archlinux.org/packages/
  - NOT_FOUND
- Gentoo: https://packages.gentoo.org/
  - NOT_FOUND
- macOS: https://formulae.brew.sh/
  - NOT_FOUND
- Alpine: https://pkgs.alpinelinux.org/packages
  - NOT_FOUND
- NixOS/nixpkgs: https://search.nixos.org/packages
  - NOT_FOUND
- openSUSE: https://build.opensuse.org/search
  - AVAILABLE

Commit:
Add python3-mlflow-pip to rosdep

n-meunier commented:
FYI @yotabits

hidmic commented on rosdep/python.yaml:
```diff
@@ -6620,6 +6620,16 @@ python3-meshio:
+      packages: [mlflow]
+  ubuntu: 
+    pip:
+      packages: [mlflow]
```
@n-meunier mind to add an entry for OSX and Gentoo? BTW it appears there's `mlflow` 1.11.0 [on OpenSUSE](https://build.opensuse.org/package/show/home:jayvdb:py-new/python-mlflow).

n-meunier replied:
Thanks for the review. I'm gonna add an entry for OSX and OpenSUSE. 
For Gentoo, I could not find the package in https://packages.gentoo.org/. Isn't it the condition for this entry?

hidmic replied:
Yeah, it doesn't look like we can use anything but `pip` on Gentoo.

n-meunier replied:
I haven't found any other example with:
```
  gentoo:
    pip:
      packages: 
```
But I added it following your comment.

tfoote replied:
My impression is that gentoo doesn't like alternative installers like pip. @allenh1  Can you confirm?

allenh1 replied:
> My impression is that gentoo doesn't like alternative installers like pip.

Yeah, you shouldn't install things with pip on gentoo. These days, if you attempt to run pip install you get an error message.

```
ERROR: (Gentoo) Please run pip with the --user option to avoid breaking python-exec
```

Previously it would just break your package manager :upside_down_face:.

What we can do, though, is create a package in the overlay repo. We've done this for a few deps. I'll make an issue there for it.

allenh1 replied:
I made a ticket to release the dependency [here](https://github.com/ros/ros-overlay/issues/1019). In the meantime, feel free to merge without a gentoo resolution, as there unfortunately isn't one.

n-meunier replied:
Thank you for the ticket and your explanation.

Commit:
Add OpenSUSE and OSX entry for mlflow

Rename package from python3-mlflow-pip to python3-mlfow since it is not only pip

Commit:
Add gentoo pip entry

hidmic reviewed (approved):
This seems reasonable to me. @wjwwood ?

Commit:
Revert "Add gentoo pip entry"

This reverts commit 3fff709370fdb1a3392592da84d80b4f13f8357e.

n-meunier commented:
@tfoote @allenh1 thank you for the review, I reverted the Gentoo entry.
